### PR TITLE
update the go version to 1.21.3

### PR DESCRIPTION
**Pull request by kmala**
Bump the go version and go runner version

### CODE REVIEW

File: .ko.yaml [hunk 1]
```diff
@@ -1 +1 @@
-defaultBaseImage: registry.k8s.io/build-image/go-runner:v2.3.1-go1.20.3-bullseye.0
+defaultBaseImage: registry.k8s.io/build-image/go-runner:v2.3.1-go1.21.3-bullseye.0
```
Comment (cartermckinnon) [inline .ko.yaml:1]: Does this image exist? Looks like it should be the `bookworm` variant: https://github.com/kubernetes/k8s.io/blob/d0222fb7613e8ea893025d92b1f93d22d10b88fa/registry.k8s.io/images/k8s-staging-build-image/images.yaml#L646

Comment (kmala) [reply .ko.yaml]: My bad, updated the tags to be proper ones

**Review by hakman — APPROVED** [review]

**Pull request merged**